### ISSUE TITLE
RenderingServer.create_local_rendering_device() memory leaks

### ISSUE THREAD

**Issue by VanessB** [godotengine/godot#71003]
### Godot version

v4.0.beta10.official

### System information

Debian GNU/Linux 11 (bullseye), Intel Core i3-8130U CPU, Mesa Intel UHD Graphics 620 (KBL GT2) 

### Issue description

Memory leaks occur while using `RenderingServer.create_local_rendering_device()`. Moreover, the engine crushes after creating many local rendering devices (even if this is done in a scoped environment, see below). If `rendering_device.free()` is called, the following warning appears: `drivers/vulkan/rendering_device_vulkan.cpp:8957 - 1 RID of type "<...>" was leaked.`, where `<...>` is any object, created with local rendering device (Shader, Texture, ...).

### Steps to reproduce

The following code reproduces the bug:

```@gdscript
extends Node

func _ready() -> void:
    for step in range(199):  # 199 times are enogh to crush the engine on my system.
        create_lrd()
        print(step)

func create_lrd() -> void:
    var rd := RenderingServer.create_local_rendering_device()

    # If something is done with rd here, it remains in memory even upon
    # return from this function.

    #rd.free() # No memory leaks occur, but Vulkan warnings are printed.
    # this line also fixes crushes.
```

### Minimal reproduction project

[RDMemoryLeak.zip](https://github.com/godotengine/godot/files/10361424/RDMemoryLeak.zip)

**Comment by wardensdev** [comment]
Creating that many local rendering devices will use up all of your video memory (per [the vulkan spec](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCreateDevice.html#_description)), resulting in a crash. The limit I can go on my machine is ~100. You have to call `RenderingDevice.free_rid(RID)` for any resources created with the rendering device manually, else they will be leaked when the RenderingDevice is freed.

I've written a small function in one of my projects to get the total memory usage for all created rendering devices to prevent too many RDs from causing a crash. Maybe having this functionality built in and documented would solve some headaches for people trying to use RenderingDevice before it gets documented.

**Comment by Chaosus** [comment]
Calling `RenderingServer.free_rid(rd)` will help maybe?

**Comment by Chaosus** [comment]
Calling `RenderingServer.free_rid(rd)` to free the local device will help maybe?

**Comment by VanessB** [comment]
> Calling `RenderingServer.free_rid(rd)` to free the local device will help maybe?

Yes, it helps, but it is still confusing that it is not done automatically. At least this should be reflected in the docs.
